jparest/src/main/java/uk/gov/homeoffice/digital/sas/jparest/validation/EntityValidator.java

```diff
@@ -41,20 +44,31 @@ public void validateAndThrowIfErrorsExist(Object objectToValidate) {
     }
   }
 
-  private static String createResourceConstraintViolationMessage(
-      Set<ConstraintViolation<Object>> constraintViolations) {
-
-    return constraintViolations.stream()
-        .collect(groupingBy(ConstraintViolation::getPropertyPath))
-        .entrySet().stream()
-        .map(entry -> {
-          var propertyErrors = entry.getValue().stream()
-              .map(ConstraintViolation::getMessage)
-              .collect(Collectors.joining(", "));
-
-          return String.format("%s has the following error(s): %s",
-              entry.getKey().toString(), propertyErrors);
-        })
-        .collect(Collectors.joining(". "));
+  private static Object[] createResourceConstraintViolationMessage(
+          Set<ConstraintViolation<Object>> constraintViolations) {
+
+    var constraintViolation = constraintViolations.iterator().next();
+    @SuppressWarnings("unchecked")
+    var hibernateConstraintViolation = constraintViolation.unwrap(
+        HibernateConstraintViolation.class
+    );
+    var payload = hibernateConstraintViolation.getDynamicPayload(ArrayList.class);
+
+    var result = constraintViolations.stream()
+            .collect(groupingBy(ConstraintViolation::getPropertyPath))
+            .entrySet().stream()
+            .map(entry -> {
+              var propertyErrors = entry.getValue().stream()
+                      .map(ConstraintViolation::getMessage)
+                      .collect(Collectors.joining(", "));
+
+              JSONObject constraintViolationErrors = new JSONObject();
```

Comment:
What is the usecase for `JSONObject`?
Could we not achieve the same thing with our own class that just had `field, message & data` members on them?
Or is `JSONObject` helping in some other way too?

Reply:
Changed to use a new class

Reply:
nice, I think we can also remove the simpleJson dependency from the pom too now

Reply:
Done :)